src/hotspot/os/aix/os_aix.cpp

```diff
@@ -448,6 +461,16 @@ static void query_multipage_support() {
       }
     }
 
+    // Can we use mmap with 64K pages? (Should be available with AIX7.3 TL1)
+    {
+      void* p = mmap(NULL, 64*K, PROT_READ | PROT_WRITE, MAP_ANON_64K | MAP_ANONYMOUS | MAP_SHARED, -1, 0);
+      assert(p != (void*) -1, "mmap failed");
+      if (p != (void*) -1) {
+        g_multipage_support.can_use_64K_mmap_pages = (64*K == os::Aix::query_pagesize(p));
+        munmap(p, 64*K);
+      }
+    }
```

Comment:
Why MAP_SHARED?

Reply:
Sorry Thomas, I oversaw your question.
I followed the the code of reserve_mmaped_memory(), where MAP_SHARED is used with the following comment:

```
  // Note: MAP_SHARED (instead of MAP_PRIVATE) needed to be able to
  // later use msync(MS_INVALIDATE) (see os::uncommit_memory).
  int flags = MAP_ANONYMOUS | MAP_SHARED;

```
I would like to have set the same flags for this test as used later on in real life.

Reply:
Okay.